From: Mari <git@cutegirl.tech>
Date: Fri, 17 Jun 2022 15:35:39 +0200
Subject: [PATCH] Manually edit OpenAPI generated code

The OpenAPI client generator for Rust generates a HashMap<String, String> as an argument,
but because it gets passed in a query string instead of e.g a JSON body it fails to generate
the correct code and instead generates a `.to_string()`, which is not available for a HashMap.

Thus we manually edit the OpenAPI generated code to call Serde so it can handle the conversion for us.

---
 api_client/src/apis/default_api.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/api_client/src/apis/default_api.rs b/api_client/src/apis/default_api.rs
index 9b40e30..a5ce421 100644
--- a/api_client/src/apis/default_api.rs
+++ b/api_client/src/apis/default_api.rs
@@ -469,7 +469,7 @@ pub async fn event_list(configuration: &configuration::Configuration, time_start
     local_var_req_builder = local_var_req_builder.query(&[("occurrence_time_start", &occurrence_time_start.to_string())]);
     local_var_req_builder = local_var_req_builder.query(&[("occurrence_time_end", &occurrence_time_end.to_string())]);
     if let Some(ref local_var_str) = labels {
-        local_var_req_builder = local_var_req_builder.query(&[("labels", &local_var_str.to_string())]);
+        local_var_req_builder = local_var_req_builder.query(&[("labels", serde_json::to_string(&local_var_str)?)]);
     }
     if let Some(ref local_var_str) = sort_by {
         local_var_req_builder = local_var_req_builder.query(&[("sort_by", &local_var_str.to_string())]);
